idea/src/org/jetbrains/kotlin/idea/inspections/UnnecessaryJavaUsageInspection.kt

```diff
@@ -28,41 +28,47 @@ import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
 class UnnecessaryJavaUsageInspection : AbstractKotlinInspection(), CleanupLocalInspectionTool {
 
     val patterns = mapOf(
-            "java.io.PrintStream.println" to "println($0)",
-            "java.io.PrintStream.print" to "print($0)",
-            "java.util.Collections.sort" to "$0.sort()"
+            "java.io.PrintStream.println" to Pair("println($0)", false),
+            "java.io.PrintStream.print" to Pair("print($0)", false),
+            "java.util.Collections.sort" to Pair("$0.sort()", false),
+            "java.util.HashMap.put" to Pair("$0[$1] = $2", true)
     )
 
     override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean, session: LocalInspectionToolSession): PsiElementVisitor {
         return object : KtVisitorVoid() {
             override fun visitQualifiedExpression(expression: KtQualifiedExpression) {
                 super.visitQualifiedExpression(expression)
 
-                val selectorExpression = expression.selectorExpression ?: return
-                if (selectorExpression !is KtCallExpression) return
-                val value = selectorExpression.valueArguments.singleOrNull() ?: return
+                val selectorExpression = expression.selectorExpression
+                if(selectorExpression !is KtCallExpression) return
+                val arguments = selectorExpression.valueArguments
+                if(arguments.isEmpty()) return
 
                 val calleeExpression = selectorExpression.calleeExpression as KtSimpleNameExpression
                 val bindingContext = calleeExpression.analyze(BodyResolveMode.PARTIAL)
                 val target = calleeExpression.mainReference.resolveToDescriptors(bindingContext).singleOrNull() ?: return
-                val pattern = target.fqNameSafe.asString()
-                if (!patterns.containsKey(pattern)) return
-
-                holder.registerProblem(expression,
-                                       "Unnecessary java usage",
-                                       ProblemHighlightType.WEAK_WARNING,
-                                       UnnecessaryJavaUsageFix(patterns[pattern]!!, value.text))
+                val pattern = patterns[target.fqNameSafe.asString()] ?: return
+                val javaUsageFix = UnnecessaryJavaUsageFix(pattern, expression.receiverExpression, arguments.map{it.text})
+                holder.registerProblem(expression, "Unnecessary java usage", ProblemHighlightType.WEAK_WARNING, javaUsageFix)
             }
         }
     }
 
-    private class UnnecessaryJavaUsageFix(val pattern: String, val value: String) : LocalQuickFix {
+    private class UnnecessaryJavaUsageFix(val pattern: Pair<String, Boolean>,
```

Comment:
Better to unpack the pair and to use separate arguments for the text and conversion mode.